tests/e2e/default/sms/africas-talking.wdio-spec.js

```diff
@@ -0,0 +1,252 @@
+const utils = require('../../../utils');
+const querystring = require('querystring');
+const commonPage = require('../../../page-objects/default/common/common.wdio.page');
+const messagesPage = require('../../../page-objects/default/sms/messages.wdio.page');
+const loginPage = require('../../../page-objects/default/login/login.wdio.page');
+const reportsPage = require('../../../page-objects/default/reports/reports.wdio.page');
+
+const CREDENTIAL_PASS = 'yabbadabbadoo';
+const CREDENTIAL_KEY = 'africastalking.com:incoming';
+
+describe('africas talking api', () => {
+  before( () => utils.saveCredentials(CREDENTIAL_KEY, CREDENTIAL_PASS) );
+
+  describe('- gateway submits new WT sms messages', () => {
+    const submitSms = body => {
+      const content = querystring.stringify(body);
+      return utils.request({
+        method: 'POST',
+        path: `/api/v1/sms/africastalking/incoming-messages?key=${CREDENTIAL_PASS}`,
+        body: content,
+        headers: {
+          'Content-Type': 'application/x-www-form-urlencoded',
+          'Content-Length': content.length,
+        },
+        json: false,
+      });
+    };
+
+    beforeEach(async () => {
+      submitSms({ from: '+64271234567', text: 'Hello', id: 'messageID' });
+      await loginPage.cookieLogin();
+    });
+
+    it('- shows content', async () => {
+      const rawNumber = '+64271234567';
+      const message = 'Hello';
+
+      await commonPage.goToMessages();
+      const { heading, summary} = await messagesPage.getMessageInListDetails(rawNumber);
+      expect(heading).to.equal(rawNumber);
+      expect(summary).to.equal(message);
+
+      await messagesPage.openMessage(rawNumber);
+
+      const { name } = await messagesPage.getMessageHeader();
+      expect(name).to.equal(rawNumber);
+
+      const { content, state, dataID } = await messagesPage.getMessageContent(1);
+      expect(content).to.equal(message);
+      expect(state).to.equal('received');
+
+      const doc = await utils.getDoc(dataID);
+      expect(doc.sms_message && doc.sms_message.gateway_ref).to.equal('messageID');
```

Comment:
```suggestion
      expect(doc.sms_message.gateway_ref).to.equal('messageID');
```

IDK if I am missing something here perhaps, but I guess this assertion will fail if `doc.sms_message` exists or not.  I think it is a lot easier to read if you do not bother with the extra code here....